packages/expo-cli/src/commands/build/ios/IOSBuilder.ts

```diff
@@ -122,14 +127,16 @@ class IOSBuilder extends BaseBuilder {
     if (!bundleIdentifier) throw missingBundleIdentifierError();
     const context = new Context();
     await context.init(this.projectDir);
-    await this.bestEffortAppleCtx(context, bundleIdentifier);
+
     await this.clearAndRevokeCredentialsIfRequested(context, { experienceName, bundleIdentifier });
```

Comment:
I was wondering if we don't need to run `bestEffortAppleCtx` before `clearAndRevokeCredentialsIfRequired`. If I understand things correctly, we need the apple context to revoke credentials on Apple Developer Portal. However, I checked the implementation (briefly, there's a lot of code there :D) of `clearAndRevokeCredentialsIfRequired` and it seems that we're prompting the user for apple id creds anyway (in the case when we don't have them yet). 
Let me know if I haven't missed anything and you've checked whether it's not necessary to have the apple context prior to calling `clearAndRevokeCredentialsIfRequested `.

Reply:
yup :) I've manually tested that case - if someone wants to revoke credentials and also wants to revoke it on the apple portal, we will ask for apple creds in a prompt.

The reason I added `bestEffortAppleCtx` is because we do full validation if we have apple creds, and limited validation otherwise. If someone doesnt have apple creds, we felt it would be a bad user experience if we asked 'do u have apple creds' every time we had a cred that we could do full validation on. Instead, we ask if u have apple creds up front so we know which checks to perform (full vs limited).

Reply:
It makes sense. Thanks for the clarification.